### PR TITLE
gives drones pacifism (except for syndrones and polymorphed drones)

### PR DESCRIPTION
**Pull request by Sishen1542**
makes it a bit harder for people to get ding dong'd as drone
also stops combat cogscarabs because thats fucking dumb

### CODE REVIEW

**Comment by GrayRachnid** [comment]
F fireaxe cogscarabs

**Comment by DeltaFire15** [comment]
F

**Comment by Trilbyspaceclone** [comment]
Make sure emaged drones loose the trait please

**Comment by dapnee** [comment]
Tiny hands. At least now when I try to throw stuff I can “lightly place” things down.

**Comment by Arturlang** [comment]
Whitelist cogscarabs, also no emagged drones on TGcode sadly @Trilbyspaceclone

**Comment by Sishen1542** [comment]
> Whitelist cogscarabs, also no emagged drones on TGcode sadly @Trilbyspaceclone

no, this is to stop combat scarabs in the first place. station drones getting it is just to make it harder for brainlets to get role banned

**Comment by Sishen1542** [comment]
updated to remove pacifism from hacked drones and re-add upon removal of hacked status

**Comment by Sishen1542** [comment]
moved everything to a var, per ghommies request

File: code/modules/mob/living/simple_animal/friendly/drone/_drone.dm [hunk 1]
```diff
@@ -100,6 +101,8 @@
 	for(var/datum/atom_hud/data/diagnostic/diag_hud in GLOB.huds)
 		diag_hud.add_to_hud(src)
 
+	if(pacifist == TRUE)
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/friendly/drone/_drone.dm]:
```suggestion
	if(pacifist)
```

File: code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm [hunk 2]
```diff
@@ -82,6 +83,7 @@
 /mob/living/simple_animal/drone/polymorphed/Initialize()
 	. = ..()
 	liberate()
+	pacifist = FALSE
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/friendly/drone/extra_drone_types.dm:86]: Move this out of initialize and into the typepath, else it just won't work and leave poly drones as pacifists.

File: code/modules/mob/living/simple_animal/friendly/drone/interaction.dm [hunk 3]
```diff
@@ -147,6 +148,7 @@
 		to_chat(src, laws)
 		to_chat(src, "<i>Having been restored, your onboard antivirus reports the all-clear and you are able to perform all actions again.</i>")
 		hacked = FALSE
+		pacifist = TRUE
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/friendly/drone/interaction.dm:151]:
```suggestion
		if(pacifist)
				ADD_TRAIT(src, TRAIT_PACIFISM, "tinyhands")
```
I wonder if multi-line code suggestions even work.

File: code/modules/mob/living/simple_animal/friendly/drone/interaction.dm [hunk 4]
```diff
@@ -132,6 +132,7 @@
 		to_chat(src, laws)
 		to_chat(src, "<i>Your onboard antivirus has initiated lockdown. Motor servos are impaired, ventilation access is denied, and your display reports that you are hacked to all nearby.</i>")
 		hacked = TRUE
+		pacifist = FALSE
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/friendly/drone/interaction.dm:135]:
```suggestion
		REMOVE_TRAIT(src, TRAIT_PACIFISM, "tinyhands")
```

**Review by Ghommie — COMMENTED** [review]
You should let cogscarabs purge those saps the team was too goofy to keep restrained on Reebe, their "cancerousness" is already counterbalanced by their density (which means bullets do NOT fly over if not specifically targetted at them) and inability to use guns.

**Comment by Ghommie** [comment]
Unresolved issues. Stale.

**Pull request closed without merging**